services/parser-database/parser.py

```diff
@@ -0,0 +1,225 @@
+    "loads/2019/09/Nuevo-Reglamento-Homologado-1.pdf",
+}
+document_list = []
+document_list.append(mty_document)
+
+
+def parse_all_documents():
+    """Parses all documents that are specified on the DB
+    (For obvious reasons, this won't work as is while the DB is not in use)
+    (But will still parse the hardcoded document)"""
+    for document in document_list:
+        file_name = document["jurisdiction"] + ".pdf"
+        download_file(document["url"], file_name)
+        print('File downloaded')
+        parse(document, file_name)
+
+
+def has_file_changed(past_hash, file_name):
+    """Sees if the file is different.
+
+    Args:
+        past_hash (string): hash to compare from
+
+    Returns:
+        [boolean]: [if the file has changed or not]
+    """
+    hasher = hashlib.sha256()
+    with open(file_name, "rb") as pdf_file:
+        file_buffer = pdf_file.read()
+        hasher.update(file_buffer)
+        if hasher.hexdigest() == past_hash:
+            return False
+        return True
+
+
+# TODO Change to point to Javier's service
+def get_keywords(text):
+    """Get keywords that relate to this article (Javier's service)
+
+    Args:
+        text (sting): text to extract keywords from
+
+    Returns:
+        [list]: list of extracted keywords
+    """
+    """
+    return requests.post(
+        "localhost:8000", params={"text": text}
+    ).json()
+    """
+    extracted_keywords = []
+    nlp_output = keywordmock.get_keywords(text)
+    json_output = json.loads(nlp_output)
+    for keyword in json_output["keywords"]:
+        extracted_keywords.append(keyword)
+    return extracted_keywords
+
+
+def article_to_dictionary(article):
+    """Converts an parsed article to a dictionary and saves it
+
+    Args:
+        article (Article): [article object]
+    """
+    article_dict = {
+        "articleNumber": article.number,
+        "text": article.text,
+        "wordCount": len(article.text.split())
+    }
+    articles_in_memory[str(article.number)] = article_dict
+    keywords = get_keywords(article.text)
+    save_keywords_in_memory(keywords, article)
+
+
+def save_keywords_in_memory(keywords, article):
+    """Saves the keywords from an article in memory
+
+    Args:
+        keywords (JSON): contains keywords
+        article (Article): article object
+    """
+    split_article = article.text.split()
+    for keyword in keywords:
+        frequency = split_article.count(keyword)
+        if keyword not in keywords_in_memory:
+            keywords_in_memory[keyword] = []
+        keywords_in_memory[keyword].append({
+            "articleNumber": article.number,
+            "frequency": frequency
+        })
+
+
+def download_file(url, filename_to_use):
+    """Downloads file from the given URL
+
+    Args:
+        url (string): [URL that contains the file to download]
+        file_name_to_use (string): [file name for file to save]
+    """
+    retriever.get_document(url, filename_to_use)
```

Comment:
What's the value of this one-line method? Can it just be inlined?

Reply:
I originally took it out so that the parser test could run with a different file. If it was hard coded the test would have to be with the mty file that may change/dissapear